File: tests/cases/walrus.py
#!/usr/bin/env python3


def show():
    # Simple walrus operator
    if (n := len([1, 2, 3])) > 2:
        print(n)

    # Walrus in while loop
    i = 0
    while (x := i * 2) < 10:
        print(x)
        i += 1


if __name__ == "__main__":
    show()

File: tests/expected/walrus.go
package main

import (
	"fmt"
)

func Show() {
	n := len([]int{1, 2, 3})
	if n > 2 {
		fmt.Printf("%v\n", n)
	}
	var i int = 0
	for true {
		var x int = (i * 2)
		if !(x < 10) {
			break
		}
		fmt.Printf("%v\n", x)
		i += 1
	}
}

func main() {
	Show()
}
